src/Data/Repositories/WalletWithdrawalRequestRepository.cs

```diff
@@ -76,6 +76,14 @@ public async Task<List<WalletWithdrawalRequest>> GetByIds(List<int> ids)
             return await applicationDbContext.WalletWithdrawalRequests.Where(wr => ids.Contains(wr.Id)).ToListAsync();
         }
 
+        public async Task<List<WalletWithdrawalRequest>> GetByReferenceIds(List<string> referenceIds)
+        {
+            await using var applicationDbContext = await _dbContextFactory.CreateDbContextAsync();
+
+            return await applicationDbContext.WalletWithdrawalRequests.Where(wr => !string.IsNullOrEmpty(wr.ReferenceId) && referenceIds.Contains(wr.ReferenceId)).ToListAsync();
```

Comment:
why `Contains` and not `Equals`?

Reply:
referenceIds is a list, you can ask for multiple referenceIds on the same call